inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,192 @@
+
+We use [testthat](https://cran.r-project.org/package=testthat).
+Contributions with test cases are easier to accept because the tests
+ensure the code does what it intends to do and nothing else. Without
+tests the maintainer needs to check the new functionality by hand, a
+burden you can lessen or remove by including tests. If you are not sure
+what parts of your code is covered by tests
+[covr](https://cran.r-project.org/package=covr) is a great tool to use
+before submission. Just run the following to get a local coverage report
+of the package so you can see exactly what lines are not covered in the
+project.
+
+``` r
+# install.packages("covr")
+co <- covr::package_coverage()
+covr::report(co)
+```
+
+Before submitting your changes, make sure that the package either still
+passes `R CMD check`, or that the warnings and/or notes have not changed
+as a result of your edits.
+
+### The Pull Request
+
+Your pull request should clearly and concisely state the motivating the
+need for change. Each change and each PR) should correspond to a branch.
+The best way to check exactly what changes you are proposing is to use
+`git diff` prior to submitting your contribution. This will ensure it
+contains only the changes necessary for the new functionality.
+
+If the PR is related to an issue, link to it in the description, with
+[the `#15`
+syntax](https://help.github.com/articles/autolinked-references-and-urls/)
+and the issue slug for context. If the PR is meant to close an issue,
+make sure one of the commit messages includes [text like `closes #44` or
+`fixes
+#101`](https://help.github.com/articles/closing-issues-using-keywords/).
+Provide the issue number and slug in the description, even if the issue
+is mentioned in the title, because auto-linking does not work in the PR
+title.
+
+  - GOOD PR title: *“Obtain user’s intent via mind-reading; fixes
+    \#86”.*
+  - BAD PR title: *“Fixes \#1043”.* Please remind us all what issue
+    \#1043 is about\!
+  - BAD PR title: *“Something about \#345”.* This will not actually
+    close issue \#345 upon merging.
+
+Add a bullet to `NEWS.md` with a concise description of the change, if
+it’s something a user would want to know when updating the package.
+[dplyr’s
+`NEWS.md`](https://github.com/tidyverse/dplyr/blob/master/NEWS.md) is a
+good source of examples. Note the sentence format, the inclusion of
+GitHub username, and links to relevant issue(s)/PR(s). We will handle
+any organization into sub-sections just prior to a release.
+
+What merits a bullet?
+
+  - Fixing a typo in the docs does not, but it is still awesome and
+    deeply appreciated.
+  - Fixing a bug or adding a new feature is bullet-worthy.
+
+When a maintainer does review a contribution, try to address the
+comments in short order, your changes are much more likely to be
+accepted if they are addressed in the next day than the next month.
+
+## On contributing
+
+### View contributing as a relationship, not a transaction
```

Comment:
I would just link to @jimhester's blog post that covers this from the Resources.